### ISSUE TITLE
PerformanceResourceTiming: domainLookupEnd

### ISSUE THREAD

**Issue by avadacatavra** [servo/servo#21260]
In components/net/http_loader.rs::http_fetch we need to set domain_lookup_end (if we're using a persistent connection). Otherwise, domain lookup should be performed by hyper, which is probably out of the scope of this issue.

On getting, the domainLookupEnd attribute MUST return as follows:

1. The same value as fetchStart, if a persistent connection [RFC7230] is used or the resource is retrieved from relevant application caches or local resources.
2. The time immediately after the user agent ends the domain data retrieval from the domain information cache, if the user agent has the domain information in cache.
3. The time immediately before the user agent finishes the domain name lookup for the resource, if the last non-redirected fetch of the resource passes the timing allow check algorithm.
4. zero, otherwise.

Spec: https://w3c.github.io/resource-timing/#dom-performanceresourcetiming-domainlookupend

**Comment by Hyperion101010** [comment]
Can you help me here , i am working on this bug !
So i did some research and found that you have struct of PerformanceResourceTiming containing domainlookup float type .
Now i have figured out how to apply all the above rules , just the part of finding the instance with struct  PerformanceResourceTiming is something that , i am not sure how can i use/find it

**Comment by Hyperion101010** [comment]
@highfive: assign me

**Comment by Hyperion101010** [comment]
I am concerned about some things here ?
It's pretty new library ,and most of part is yet to be implemented .But i'll try my best to work upon it silently

**Comment by jdm** [comment]
Given that hyper does not expose a callback for letting us know if we're using a persistent connection or performing domain resolution, I do not believe we can implement this without modifying hyper.

**Comment by Hyperion101010** [comment]
avadacatavara has commented that hyper is out of this scope , but still for a full implementation hyper support will be needed, which i think we can add in incremental way.

**Comment by jdm** [comment]
Yes, it's probably still worthwhile to expose the DOM API and use the same value as responseStart.
